### PR TITLE
updated tests to use pytest fixtures, and removed redundant test

### PR DESCRIPTION
**Pull request by sherwoodf**
Small PR addressing liviu's comment on my last one: https://github.com/BioImage-Archive/bia-exporter/pull/8/

Just updating the tests to use pytest fixtures rather than calling the util functions directly. I've kept them as separate functions just in case we need to use them while passing in UUIDs to make the tests work.

Removed the first test i wrote as i realised it's basically testing the same code, except that it avoids making API calls by using an example that wouldn't cause calls to our api (so really it's just testing the mapping method)

### CODE REVIEW

**Review by kbab — COMMENTED** [review]
It all looks good to me. However, I had to add `pytest-mock` to be able to run the tests. Does the `pyproject.toml` file need to be updated?

**Comment by sherwoodf** [comment]
> It all looks good to me. However, I had to add `pytest-mock` to be able to run the tests. Does the `pyproject.toml` file need to be updated?

Interesting... my pyproject.toml does not include pytest-mock, but pytest-mock is included in the lock file dependencies. However, my toml file does not match the one in github. VSCode & git aren't telling me i made any changes & the .toml file isn't in the gitignore so i would expect it to have shown up... will try to work out what's happening...

**Comment by sherwoodf** [comment]
Okay i've double checked and i've made no changes to the toml file, and the lock file generates with the required dependies including pytest-mock, so i don't know why you needed to add that. Did you regenerate the lockfile before running the tests?

**Comment by kbab** [comment]
I did a git pull and tried to run the tests. I got the following error:
```
ImportError while importing test module '/home/kola/code/embl_projects/BioImageArchive/bia-exporter/tests/test_data_mapping_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/home/kola/.asdf/installs/python/3.11.7/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_data_mapping_utils.py:1: in <module>
    from pytest_mock import mocker
E   ModuleNotFoundError: No module named 'pytest_mock'

```
I then did `poetry install --sync` which I believe updates my environment according to pyproject.toml? but still got the error. The error only stopped after I did `poetry add pytest-mock` and now my pyproject.toml has `pytest-mock = "^3.14.0"` . However, this does not stop you committing. If anyone else has issues at least we know what to do.

**Comment by sherwoodf** [comment]
Ah okay, that's my bad then - i had not been using sync, so clearly had installed it in the past & then removed it, but it had stayed in my env. Will update the toml now.

File: .github/workflows/exporter.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (liviuba) [inline .github/workflows/exporter.yaml:24]: Doing a poetry install is a good idea, but the pip upgrade & poetry install (as in, installing poetry) would just add more time and possibly make the build runs non(/less) deterministic.

Maybe use something similar to this https://github.com/BioImage-Archive/bia-integrator/blob/main/.github/workflows/core.yaml#L21 with a fixed python/poetry version instead?

**Review by liviuba — APPROVED** [review]
LGTM! Please see comment

**Pull request merged**